CommandLine/emake/EnigmaPlugin.cpp

```diff
@@ -29,23 +29,28 @@ int EnigmaPlugin::Init()
 #if CURRENT_PLATFORM_ID == OS_WINDOWS
 #	define dlopen(x, y) LoadLibrary(x)
   std::string extension = ".dll";
-#elseif CURRENT_PLATFORM_ID ==  OS_MACOSX
+  std::string prefix = "";
+#elif CURRENT_PLATFORM_ID ==  OS_MACOSX
   std::string extension = ".dylib";
+  std::string prefix = "lib";
 #else
   std::string extension = ".so";
+  std::string prefix = "lib";
 #endif
 
-  _handle = dlopen(("./libcompileEGMf" + extension).c_str(), RTLD_LAZY);
+  std::string pluginName = "./" + prefix + "compileEGMf" + extension;
+  
+  _handle = dlopen(pluginName.c_str(), RTLD_LAZY);
 
   if (!_handle)
   {
-    std::cerr << "Error Loading Plugin" << std::endl;
+    std::cerr << "Error Loading Plugin " << pluginName << std::endl;
```

Comment:
Nit: Better to quote names, in case they're empty.

Reply:
This is fine for now because he's only loading a single plugin, just remember to do it in the future.